libraries/AP_BattMonitor/AP_BattMonitor_Backend.h

```diff
@@ -85,13 +85,18 @@ class AP_BattMonitor_Backend
     // return mavlink fault bitmask (see MAV_BATTERY_FAULT enum)
     virtual uint32_t get_mavlink_fault_bitmask() const { return 0; }
 
-    // logging functions 
+    // logging functions
     void Log_Write_BAT(const uint8_t instance, const uint64_t time_us) const;
     void Log_Write_BCL(const uint8_t instance, const uint64_t time_us) const;
 
     // set desired MPPT powered state (enabled/disabled)
     virtual void mppt_set_powered_state(bool power_on) {};
 
+    // returns true if the battery can be shutdown with shutdown()
+    virtual bool can_shutdown() { return false; };
+    // shuts the battery down if supported
+    virtual void shutdown() {};
```

Comment:
Could these be combined?

Reply:
The reason I separated those is so that we could check if a battery could be shutdown without actually trying to shut it down. Thus, if we for some reason had one battery monitor that supported shutdown and another that didnt, we wouldn't allow shutting down of the system. What are your thoughts there? Maybe a "shutdown" should attempt to shut down everything that supports it and leave the others online?